examples/shared_stream_controllers.rs

```diff
@@ -0,0 +1,234 @@
+use std::{sync::Arc, time::Duration};
+
+use futures::StreamExt;
+use k8s_openapi::api::core::v1::{Pod, PodCondition};
+use kube::{
+    api::{Patch, PatchParams},
+    runtime::{controller::Action, reflector, watcher, Config, Controller, WatchStreamExt},
+    Api, Client, ResourceExt,
+};
+use tokio::sync::mpsc;
+use tracing::{info, warn};
+
+use thiserror::Error;
+
+// Helper module that namespaces two constants describing a Kubernetes status condition
+pub mod condition {
+    pub static UNDOCUMENTED_TYPE: &str = "UndocumentedPort";
+    pub static STATUS_TRUE: &str = "True";
+}
+
+const SUBSCRIBE_BUFFER_SIZE: usize = 256;
+
+#[derive(Debug, Error)]
+enum Error {
+    #[error("Failed to patch pod: {0}")]
+    WriteFailed(#[source] kube::Error),
+
+    #[error("Missing po field: {0}")]
+    MissingField(&'static str),
+}
+
+#[derive(Clone)]
+struct Data {
+    client: Client,
+}
+
+/// A simple reconciliation function that will copy a pod's labels into the annotations.
+async fn reconcile_metadata(pod: Arc<Pod>, ctx: Arc<Data>) -> Result<Action, Error> {
+    if pod.name_any() == "kube-system" {
+        return Ok(Action::await_change());
+    }
+
+    let labels = pod.labels();
+    if labels.is_empty() {
+        return Ok(Action::await_change());
+    }
+
+    let mut annotations = pod.annotations().clone();
+    for (key, value) in labels {
+        annotations.insert(key.to_owned(), value.to_owned());
+    }
```

Comment:
this feels like a [`BTreeMap::append`](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.append)

```suggestion
    // combine labels and annotations into a new map
    let mut map = pod.labels_mut();
    pod.annotations_mut().append(map);
```

Reply:
Hm, ended up backing out of the change since `append` requires both caller and arg to be a mutable reference. Since we operate on the same pod, the compiler won't allow us to do this (unless we use a `RefCell` or something which is going to make this a bit more complicated than it has to be).

I'll just use `extend` on it with a cloned set of labels.